File: models/weight_log.py
from datetime import datetime
from database import db

class WeightLog(db.Model):
    __tablename__ = 'weight_logs'
    
    id = db.Column(db.Integer, primary_key=True)
    member_id = db.Column(db.Integer, db.ForeignKey('family_members.id'), nullable=False)
    weight_kg = db.Column(db.Float, nullable=False)
    bmi = db.Column(db.Float, nullable=False)
    logged_date = db.Column(db.Date, default=datetime.utcnow().date)
    
    def to_dict(self):
        return {
            'id': self.id,
            'member_id': self.member_id,
            'weight_kg': self.weight_kg,
            'bmi': self.bmi,
            'logged_date': self.logged_date.strftime('%Y-%m-%d')
        }

File: models/weightLog.js
const { DataTypes, Model } = require('sequelize');
const sequelize = require('../database');

class WeightLog extends Model {
  toDict() {
    return {
      id: this.id,
      member_id: this.member_id,
      weight_kg: this.weight_kg,
      bmi: this.bmi,
      logged_date: this.logged_date
    };
  }
}

WeightLog.init({
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true
  },
  member_id: {
    type: DataTypes.INTEGER,
    allowNull: false
  },
  weight_kg: {
    type: DataTypes.FLOAT,
    allowNull: false
  },
  bmi: {
    type: DataTypes.FLOAT,
    allowNull: false
  },
  logged_date: {
    type: DataTypes.DATEONLY,
    defaultValue: DataTypes.NOW
  }
}, {
  sequelize,
  modelName: 'WeightLog',
  tableName: 'weight_logs',
  timestamps: false
});

module.exports = WeightLog;
